addons/event_sale/models/sale_order.py

```diff
@@ -1,53 +1,45 @@
 # -*- coding: utf-8 -*-
 
-from openerp import api
-from openerp.osv import fields, osv
+from openerp import api, fields, models
 
 
-class sale_order(osv.osv):
+class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    def action_button_confirm(self, cr, uid, ids, context=None):
-        # TDE note: This method works on a list of one id (see sale/sale.py) so working on ids[0] seems safe.
-        res = super(sale_order, self).action_button_confirm(cr, uid, ids, context=context)
-        redirect_to_event_registration = any(line.event_id for order in self.browse(cr, uid, ids, context=context) for line in order.order_line)
-        if redirect_to_event_registration:
-            event_ctx = dict(context, default_sale_order_id=ids[0])
-            return self.pool.get('ir.actions.act_window').for_xml_id(cr, uid, 'event_sale', 'action_sale_order_event_registration', event_ctx)
-        else:
-            return res
+    @api.multi
+    def action_button_confirm(self):
+        self.ensure_one()
+        super(SaleOrder, self).action_button_confirm()
+        if self.mapped('order_line.event_id'):
+            return self.env['ir.actions.act_window'].with_context(default_sale_order_id=self.id).for_xml_id('event_sale', 'action_sale_order_event_registration')
```

Comment:
Is it normal that the result of the `super()` call is ignored and nothing is returned if tehere are no `order_line.event_id`?